bmc-log-collector/log-collector.go

```diff
@@ -90,66 +87,75 @@ func (c *selCollector) collectSystemEventLog(ctx context.Context, m Machine, log
 
 		err = updateLastPointer(lastPtr, filePath)
 		if err != nil {
-			slog.Error("failed to write a pointer file.", "err", err, "serial", m.Serial, "createUnixtime", createUnixtime, "LastReadId", lastId, "ptrDir", c.ptrDir)
+			slog.Error("failed to write a pointer file.", "err", err, "serial", m.Serial, "ptrDir", c.ptrDir)
 		}
 		return
 	}
 
 	// Increment the success counter
 	counterRequestSuccess.WithLabelValues(m.Serial).Inc()
 
-	var members RedfishJsonSchema
-	if err := json.Unmarshal(byteJSON, &members); err != nil {
+	var response RedfishJsonSchema
+	if err := json.Unmarshal(byteJSON, &response); err != nil {
 		slog.Error("failed to translate JSON to go struct.", "err", err, "serial", m.Serial, "ptrDir", c.ptrDir)
 		return
 	}
 
-	for i := len(members.Sel) - 1; i >= 0; i-- {
-		t, _ := time.Parse(layout, members.Sel[i].Create)
-		createUnixtime = t.Unix()
-		lastId, _ = strconv.Atoi(members.Sel[i].Id)
-		members.Sel[i].Serial = m.Serial
-		members.Sel[i].BmcIP = m.BmcIP
-		members.Sel[i].NodeIP = m.NodeIP
-
-		if lastPtr.LastReadId < lastId {
-			// Normal case without iDRAC log reset.
-			bmcByteJsonLog, err := json.Marshal(members.Sel[i])
+	createTime, err := time.Parse(time.RFC3339, response.Sel[len(response.Sel)-1].Create)
+	if err != nil {
+		slog.Error("failed to parse for time", "err", err, "serial", m.Serial)
+		return
+	}
+	firstCreateTime := createTime.Unix()
+
+	for i := len(response.Sel) - 1; i >= 0; i-- {
+		currentId, err := strconv.Atoi(response.Sel[i].Id)
+		if err != nil {
+			slog.Error("failed to strconv", "err", err, "serial", m.Serial, "LastReadId", currentId, "ptrDir", c.ptrDir)
+			continue
+		}
+		// Add the information to identify of the node
+		response.Sel[i].Serial = m.Serial
+		response.Sel[i].BmcIP = m.BmcIP
+		response.Sel[i].NodeIP = m.NodeIP
+
+		if lastPtr.LastReadId < currentId {
+			bmcByteJsonLog, err := json.Marshal(response.Sel[i])
 			if err != nil {
-				slog.Error("failed to marshal the system event log", "err", err, "serial", m.Serial, "createUnixtime", createUnixtime, "LastReadId", lastId, "ptrDir", c.ptrDir)
+				slog.Error("failed to marshal the system event log", "err", err, "serial", m.Serial, "lastPtr.LastReadId", lastPtr.LastReadId, "currentLastReadId", currentId, "ptrDir", c.ptrDir)
 			}
+
 			err = logWriter.write(string(bmcByteJsonLog), m.Serial)
 			if err != nil {
-				slog.Error("failed to output log", "err", err, "serial", m.Serial, "createUnixtime", createUnixtime, "LastReadId", lastId, "ptrDir", c.ptrDir)
+				slog.Error("failed to output log", "err", err, "serial", m.Serial, "bmcByteJsonLog", string(bmcByteJsonLog), "currentLastReadId", currentId, "ptrDir", c.ptrDir)
 			}
-			lastPtr.LastReadId = lastId
-			lastPtr.LastReadTime = createUnixtime
+
+			lastPtr.LastReadId = currentId
+			lastPtr.LastReadTime = firstCreateTime
```

Comment:
Sorry for the additional comment.
`LastReadTime` is no longer needed, isn't it?

Reply:
sure, it is not using variable.
deleted